### PR TITLE
ECOM-5031 create the serialization for programs api

### PR DESCRIPTION
**Pull request by schenedx**
@rlucioni Please review
The PR creates the desired API for exposing programs data
Please note this PR is against @clintonb's program data model branch. Once the model branch is merged into master, I will update the PR base
The example program data this serialization provides is at https://gist.github.com/schenedx/9c8fa5b365e47c7795c6794b89b0afe1

### CODE REVIEW

**Comment by clintonb** [comment]
👍

File: course_discovery/apps/api/tests/test_serializers.py [hunk 1]
```diff
@@ -182,7 +185,11 @@ def test_data(self):
             'marketing_slug': program.marketing_slug,
             'marketing_url': program.marketing_url,
             'card_image_url': program.card_image_url,
+            'banner_image_url': program.banner_image_url,
+            'authoring_organizations': OrganizationSerializer(program.authoring_organizations, many=True).data,
+            'courses': CourseSerializer(program.courses, many=True, context={'request': request}).data,
```
Comment (rlucioni) [inline course_discovery/apps/api/tests/test_serializers.py:190]: Is filtering of excluded runs meant to happen as part of this?

**Comment by rlucioni** [comment]
@schenedx 👍 , though we still need to filter excluded runs from the response to maintain parity with the existing API.

File: course_discovery/apps/api/tests/test_serializers.py (hunk 1 above)
Comment (schenedx) [inline course_discovery/apps/api/tests/test_serializers.py:190]: Good catch. I'll merge this and work on the exclusion next with a new PR

**Pull request merged**